app/controllers/planning_applications/validation/document/redactions_controller.rb

```diff
@@ -34,7 +35,17 @@ def planning_application_params
         end
 
         def documents_params
-          %i[file redacted publishable validated]
+          [:file, :redacted, :publishable, :validated, tags: []]
+        end
+
+        def filter_tags!
+          return unless params[:planning_application][:documents_attributes]
+
+          params[:planning_application][:documents_attributes].each do |index, document_attributes|
+            if document_attributes[:tags].present? && document_attributes[:tags].is_a?(String)
+              params[:planning_application][:documents_attributes][index][:tags] = document_attributes[:tags].split
+            end
+          end
```

Comment:
I'm confused by the format of `documents_attributes` — is it a hash containing other hashes? And sometimes `tags` is a string and sometimes it's an array, so we have to convert the strings to arrays?

Reply:
Btw `present?` might be unnecessary here, because `nil.is_a?(String)` is false:
```suggestion
          params[:planning_application][:documents_attributes].each do |index, document_attributes|
            if document_attributes[:tags].is_a?(String)
              params[:planning_application][:documents_attributes][index][:tags] = document_attributes[:tags].split
            end
          end
```